sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -404,13 +439,20 @@ private async Task RunBatchReceiveLoopAsync(CancellationToken cancellationToken)
                     {
                         try
                         {
-                            receiver = await sessionClient.AcceptNextSessionAsync(
+                            ServiceBusSessionReceiver sessionReceiver = await sessionClient.AcceptNextSessionAsync(
                                 _entityPath,
                                 new ServiceBusSessionReceiverOptions
                                 {
                                     PrefetchCount = _serviceBusOptions.PrefetchCount
                                 },
                                 cancellationToken).ConfigureAwait(false);
+
+                            receiver = sessionReceiver;
```

Comment:
Curious - why did we change the assignment to be two steps?

Reply:
I initially took a different approach with the session logic, and needed to get the session id for the message cache. Then I changed the way I was handling sessions so I can switch this back as well.